src/server/services/communities.js

```diff
@@ -66,45 +68,53 @@ export default class Communities {
      * groups, assuming they are up-to-date. */
     this.private.getMetadata = (communityId, knownGroups = {}) =>
       new Promise((resolve, reject) => {
-        const uri = path.resolve(__dirname, '../tc-communities',
-          communityId, 'metadata.json');
+        /* Metadata itself, at the moment, are read from configuration files.
+         * And in-memory data will be used if already exists */
+
+        let metadata;
+        if (COMMUNITY_META_DATA[communityId]) {
+          metadata = COMMUNITY_META_DATA[communityId];
+        } else {
+          const uri = path.resolve(__dirname, '../tc-communities',
+            communityId, 'metadata.json');
 
-        /* Metadata itself, at the moment, are read from configuration files. */
-        fs.readFile(uri, 'utf8', (err, res) => {
-          if (err) {
+          metadata = fs.readFileSync(uri, 'utf8');
```

Comment:
This is bad for two reasons:
1. Sync IO operation at server has impact on the performance for everybody. Sure, these reads happen only once for each community after each restart of the app, but still, let's follow the best practice and do it async.
2. They it was done before, if a non-existing community ID was passed in, the read operation failed, and the correct error message was generated. With your change, if `fs.readFileSync(..)` fails it will just crush the code (sure, we could wrap the call into `try {} catch(e) {}`, but see n1).

Reply:
I'll fix this...